extensions/parquet-extensions/src/main/java/io/druid/data/input/parquet/DimensionFieldConverter.java

```diff
@@ -0,0 +1,189 @@
+    public void addBoolean(boolean value)
+    {
+      parent.addDimensionMetric(field, value + "");
+    }
+
+    @Override
+    public void addInt(int value)
+    {
+      parent.addDimensionMetric(field, value + "");
+    }
+
+    @Override
+    public void addLong(long value)
+    {
+      parent.addDimensionMetric(field, value + "");
+    }
+
+  }
+
+  /**
+   * enable string field read from dictionary , (refer to parquet-pig)
+   *
+   * @author du00
+   */
+  public static class StringFieldConverter extends PrimitiveConverter
+  {
+    private DruidGroupConverter parent;
+    private String field;
+
+    private boolean dictionarySupport;
+    private String[] dict;
+
+    public StringFieldConverter(DruidGroupConverter parent, String field)
+    {
+      this.parent = parent;
+      this.field = field;
+    }
+
+    @Override
+    final public void addBinary(Binary value)
+    {
+      parent.addDimensionMetric(field, value.toStringUsingUTF8());
+    }
+
+    @Override
+    public boolean hasDictionarySupport()
+    {
+      return dictionarySupport;
+    }
+
+    @Override
+    public void setDictionary(Dictionary dictionary)
+    {
+      dict = new String[dictionary.getMaxId() + 1];
+      for (int i = 0; i <= dictionary.getMaxId(); i++) {
+        dict[i] = dictionary.decodeToBinary(i).toStringUsingUTF8();
+      }
+      dictionarySupport = true;
+    }
+
+    @Override
+    public void addValueFromDictionary(int dictionaryId)
+    {
+      parent.addDimensionMetric(field, dict[dictionaryId]);
+    }
+  }
+
+  public static class ListFieldConverter extends DruidGroupConverter
+  {
+
+    private final Converter[] converters;
+    private DruidGroupConverter parent;
+    private String field;
+
+    public ListFieldConverter(DruidGroupConverter parent, GroupType schema, String name)
+    {
+      this.parent = parent;
+      this.field = name;
+      this.converters = new Converter[schema.getFieldCount()];
+
+      int i = 0;
+      for (Type field : schema.getFields()) {
+        if (field.isPrimitive()) {
+          if (field.getOriginalType() == OriginalType.UTF8) {
+            converters[i++] = new StringFieldConverter(this, null);
+          } else {
+            converters[i++] = new NonStringFieldConverter(this, null);
+          }
+        } else {
+          converters[i++] = new ListFieldConverter(this, field.asGroupType(), name);
+        }
+      }
+    }
+
+
+    private List<Object> list;
+
+    /**
+     * called at initialization based on schema
+     * must consistently return the same object
+     *
+     * @param fieldIndex index of the field in this group
+     *
+     * @return the corresponding converter
+     */
+    @Override
+    public Converter getConverter(int fieldIndex)
+    {
+      return converters[fieldIndex];
+    }
+
+    /**
+     * called at the beginning of the group managed by this converter
+     */
+    @Override
+    public void start()
+    {
+      list = Lists.newArrayList();
```

Comment:
you can just define:
private final List<Object> columns = Lists.newArrayList();